src/components/GraphFilter/DurationDropdown.tsx

```diff
@@ -24,27 +25,43 @@ export class DurationDropdown extends React.Component<DurationDropdownProps, Dur
     [604800, '7 days'],
     [2592000, '30 days']
   ];
+  static InitialLabel = 'Select Duration';
+
   constructor(props: DurationDropdownProps) {
     super(props);
     this.state = {
-      currentDuration: props.initialDuration
+      currentDuration: props.initialDuration,
+      currentDurationName: this.getDurationName(props.initialDuration)
     };
   }
 
+  getDurationName = (duration: number) => {
```

Comment:
Instead of having a lookup function to retrieve the name from the duration, you could directly have the tuple [duration, name] as key of `MenuItem`, it would make it a bit simpler.